src/compas/geometry/_primitives/cylinder.py

```diff
@@ -0,0 +1,293 @@
+    # ==========================================================================
+
+    def copy(self):
+        """Makes a copy of this ``Cylinder``.
+
+        Returns
+        -------
+        Cylinder
+            The copy.
+
+        """
+        cls = type(self)
+        return cls(self.plane.copy(), self.radius, self.height)
+
+    # ==========================================================================
+    # transformations
+    # ==========================================================================
+
+    def transform(self, transformation):
+        """Transform the cylinder.
+
+        Parameters
+        ----------
+        transformation : :class:`Transformation`
+            The transformation used to transform the cylinder.
+
+        Examples
+        --------
+        >>> from compas.geometry import Frame
+        >>> from compas.geometry import Transformation
+        >>> from compas.geometry import Plane
+        >>> from compas.geometry import Cylinder
+        >>> cylinder = Cylinder(Plane.worldXY(), 5, 7)
+        >>> frame = Frame([1, 1, 1], [0.68, 0.68, 0.27], [-0.67, 0.73, -0.15])
+        >>> T = Transformation.from_frame(frame)
+        >>> cylinder.transform(T)
+
+        """
+        self.plane.transform(transformation)
+
+    def transformed(self, transformation):
+        """Returns a transformed copy of the current cylinder.
+
+        Parameters
+        ----------
+        transformation : :class:`Transformation`
+            The transformation used to transform the cylinder.
+
+        Returns
+        -------
+        :class:`cylinder`
+            The transformed cylinder.
+
+        Examples
+        --------
+        >>> from compas.geometry import Frame
+        >>> from compas.geometry import Transformation
+        >>> from compas.geometry import Plane
+        >>> from compas.geometry import Cylinder
+        >>> cylinder = Cylinder(Plane.worldXY(), 5, 7)
+        >>> frame = Frame([1, 1, 1], [0.68, 0.68, 0.27], [-0.67, 0.73, -0.15])
+        >>> T = Transformation.from_frame(frame)
+        >>> circle_transformed = cylinder.transformed(T)
+
+        """
+        cylinder = self.copy()
+        cylinder.transform(transformation)
+        return cylinder
+
+
+# ==============================================================================
+# Main
+# ==============================================================================
+
+if __name__ == "__main__":
+    from compas.geometry import Frame
+    from compas.geometry import Transformation
+    cylinder = Cylinder(Plane.worldXY(), 5, 7)
+    frame = Frame([1, 1, 1], [0.68, 0.68, 0.27], [-0.67, 0.73, -0.15])
+    print(frame.normal)
+    T = Transformation.from_frame(frame)
+    cylinder.transform(T)
+    print(cylinder)
+
+    print(Plane.worldXY().data)
+    data = {'plane': Plane.worldXY().data, 'radius': 5., 'height': 7.}
+    cylinder = Cylinder.from_data(data)
+    print(cylinder)
+
+    import doctest
+    doctest.testmod()
```

Comment:
Like that you pass the imports to the examples
```suggestion
    doctest.testmod(globs=globals())
```

Reply:
I *can* add this, but as I copied most of the new cylinder class from other existing primitives, it would have to be changed there as well in order to be coherent?

Reply:
I don't think this needs to be coherent, it is only practical to make example strings shorter: sometimes needed, sometimes not